src/v/archival/archival_metadata_stm.cc

```diff
@@ -1018,81 +946,94 @@ ss::future<> archival_metadata_stm::apply(const model::record_batch& b) {
+                }
 
-            switch (key) {
-            case add_segment_cmd::key:
-                apply_add_segment(
-                  serde::from_iobuf<add_segment_cmd::value>(r.release_value()));
-                break;
-            case truncate_cmd::key:
-                apply_truncate(
-                  serde::from_iobuf<truncate_cmd::value>(r.release_value()));
-                break;
-            case update_start_offset_cmd::key:
-                apply_update_start_offset(
-                  serde::from_iobuf<update_start_offset_cmd::value>(
-                    r.release_value()));
-                break;
-            case cleanup_metadata_cmd::key:
-                apply_cleanup_metadata();
-                break;
-            case mark_clean_cmd::key:
-                apply_mark_clean(
-                  serde::from_iobuf<mark_clean_cmd::value>(r.release_value()));
-                break;
-            case truncate_archive_init_cmd::key:
-                apply_truncate_archive_init(
-                  serde::from_iobuf<truncate_archive_init_cmd::value>(
-                    r.release_value()));
-                break;
-            case truncate_archive_commit_cmd::key: {
-                auto cmd
-                  = serde::from_iobuf<truncate_archive_commit_cmd::value>(
-                    r.release_value());
-                apply_truncate_archive_commit(
-                  cmd.start_offset, cmd.bytes_removed);
-            } break;
-            case update_start_kafka_offset_cmd::key:
-                apply_update_start_kafka_offset(
-                  serde::from_iobuf<update_start_kafka_offset_cmd::value>(
-                    r.release_value()));
-                break;
-            case reset_metadata_cmd::key:
-                apply_reset_metadata();
-                break;
-            case spillover_cmd::key:
-                apply_spillover(
-                  serde::from_iobuf<spillover_cmd>(r.release_value()));
-                break;
-            case replace_manifest_cmd::key:
-                apply_replace_manifest(r.release_value());
-                break;
-            case process_anomalies_cmd::key:
-                apply_process_anomalies(r.release_value());
-                break;
-            case reset_scrubbing_metadata::key:
-                apply_reset_scrubbing_metadata();
-                break;
-            case update_highest_producer_id_cmd::key:
-                apply_update_highest_producer_id(
-                  serde::from_iobuf<update_highest_producer_id_cmd::value>(
-                    r.release_value()));
-                break;
-            default:
-                throw std::runtime_error(fmt_with_ctx(
-                  fmt::format,
-                  "Unknown archival metadata STM command {}",
-                  static_cast<int>(key)));
-            };
-        });
+                switch (key) {
+                case add_segment_cmd::key:
```

Comment:
formatting in a separate commit?

Reply:
Formatting changed due to extra indent level from try-catch bock. I don't understand why this need to be a separate commit. It's part of the change. All this code uses different exception handling approach now and it's visible in the commit history.

Reply:
>I don't understand why this need to be a separate commit. It's part of the change.

It doesn't have to be, but the importance of making commits easy to review increases with the size/complexity of the commit itself. You could add that try-catch block as a commit before this one as a mechanical change. Then it would lower the cognitive complexity of the next commit. It'd be a silly commit, but it would lower complexity.